numpy/lib/function_base.py

```diff
@@ -4014,122 +4135,234 @@ def _quantile_is_valid(q):
-
-    if axis is None:
-        axis = 0
+            arr = a.copy()
+    result = _quantile(arr,
+                       quantiles=q,
+                       axis=axis,
+                       interpolation=interpolation,
+                       out=out)
+    if q.ndim == 0:
+        result = np.squeeze(result, axis=0)
+    if result.ndim == 0 and out is None:
+        result = result[()]  # unpack 0d arrays
+    elif result.size == 1 and out is None and np.isscalar(q):
+        result = result[0]
+    return result
 
-    if q.ndim > 2:
-        # The code below works fine for nd, but it might not have useful
-        # semantics. For now, keep the supported dimensions the same as it was
-        # before.
-        raise ValueError("q must be a scalar or 1d")
-
-    Nx = ap.shape[axis]
-    indices = not_scalar(q * (Nx - 1))
-    # round fractional indices according to interpolation method
-    if interpolation == 'lower':
-        indices = floor(indices).astype(intp)
-    elif interpolation == 'higher':
-        indices = ceil(indices).astype(intp)
-    elif interpolation == 'midpoint':
-        indices = 0.5 * (floor(indices) + ceil(indices))
-    elif interpolation == 'nearest':
-        indices = around(indices).astype(intp)
-    elif interpolation == 'linear':
-        pass  # keep index as fraction and interpolate
-    else:
-        raise ValueError(
-            "interpolation can only be 'linear', 'lower' 'higher', "
-            "'midpoint', or 'nearest'")
 
+def _quantile(
+        arr: np.array,
+        quantiles: np.array,
+        axis: int = -1,
+        interpolation=QuantileInterpolation.INCLUSIVE,
+        out=None,
+):
+    """
+    Private function that doesn't support extended axis or keepdims.
+    These methods are extended to this function using _ureduce
+    See nanpercentile for parameter usage
+    It computes the quantiles of the array for the given axis.
+    A linear interpolation is performed based on the interpolation_type.
+
+    By default, the interpolation is QuantileInterpolation.INCLUSIVE where
+    alpha == beta == 1 which performs the 7th method of Hyndman&Fan.
+    With QuantileInterpolation.MEDIAN_UNBIASED we get  alpha == beta == 1/3
+    thus the 8th method of Hyndman&Fan.
+    """
+    # --- Setup
+    # ufuncs cause 0d array results to decay to scalars (see gh-13105), which
+    # makes them problematic for __setitem__ and attribute access. As a
+    # workaround, we call this on the result of every ufunc on a possibly-0d
+    # array.
+    not_scalar = np.asanyarray
+    values_count = arr.shape[axis]
+    if values_count == 0:
+        return np.NAN
+    if values_count == 1:
+        result = np.take(arr, indices=0, axis=axis)
+        if np.issubdtype(arr.dtype, np.inexact) and np.isnan(result):
+            warnings.warn("All-NaN slice encountered", RuntimeWarning,
+                          stacklevel=4)
+        return np.broadcast_to(result, (quantiles.size,) + result.shape)
```

Comment:
```
 >           return np.broadcast_to(result, (quantiles.size,) + result.shape)
E           AttributeError: 'int' object has no attribute 'shape'
```
This can fail if `arr` is an object array, as `result` is thus not guaranteed to have the `shape` attribute.

Reply:
Good catch, I thought arr was wrapped in `asarray` above, but I might have lost it at some point.